src/banks/registries/directory.py

```diff
@@ -38,24 +44,69 @@ def _load(self):
+            pf = PromptFile(name=name, version=version, path=path, meta={})
             self._index.files.append(pf)
         self._index_path.write_text(self._index.model_dump_json())
 
-    def get(self, *, name: str, version: str | None = None) -> "Prompt":
-        version = version or "0"
+    def get_prompt(self, *, name: str, version: str | None = None) -> "Prompt":
+        version = version or DEFAULT_VERSION
         for pf in self._index.files:
             if pf.name == name and pf.version == version and pf.path.exists():
                 return Prompt(pf.path.read_text())
         raise TemplateNotFoundError
 
-    def set(self, *, name: str, prompt: Prompt, version: str | None = None, overwrite: bool = False):
-        version = version or "0"
+    def _get_prompt_file(self, *, name: str, version: str) -> PromptFile | None:
         for pf in self._index.files:
-            if pf.name == name and pf.version == version and overwrite:
-                pf.path.write_text(prompt.raw)
-                return
-        new_prompt_file = self._path / "{name}.{version}.jinja"
+            if pf.name == name and pf.version == version:
+                return pf
+        return None
+
+    def _create_pf(  # pylint: disable=too-many-arguments
+        self, *, name: str, prompt: Prompt, version: str, overwrite: bool, meta: dict
+    ) -> "PromptFile":
+        pf = self._get_prompt_file(name=name, version=version)
+        if pf:
+            if not overwrite:
+                msg = f"Prompt {name}.{version}.jinja already exists. Use overwrite=True to overwrite."
+                raise ValueError(msg)
+            pf.path.write_text(prompt.raw)
+            pf.meta = meta
+            return pf
+        new_prompt_file = self._path / f"{name}.{version}.jinja"
         new_prompt_file.write_text(prompt.raw)
-        pf = PromptFile(name=name, version=version, path=new_prompt_file)
-        self._index.files.append(pf)
+        pf = PromptFile(name=name, version=version, path=new_prompt_file, meta=meta)
+        return pf
+
+    def set(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        name: str,
+        prompt: Prompt,
+        meta: dict | None = None,
+        version: str | None = None,
+        overwrite: bool = False,
+    ):
+        version = version or DEFAULT_VERSION
+        meta = {**(meta or {}), "created_at": time.ctime()}
+
+        pf = self._create_pf(name=name, prompt=prompt, version=version, overwrite=overwrite, meta=meta)
+        if pf not in self._index.files:
+            self._index.files.append(pf)
+        self._index_path.write_text(self._index.model_dump_json())
+
+    def get_meta(self, *, name: str, version: str | None = None) -> dict:
+        version = version or DEFAULT_VERSION
+        for pf in self._index.files:
+            if pf.name == name and pf.version == version:
+                return pf.meta
+        raise MetaNotFoundError
```

Comment:
Not sure about this, the error will be triggered by the fact the requested `pf` was not found in the index... Also what do you want to signal with `MetaNotFound`, the absence of the dict or an emty dict?

Reply:
It was like that before. I changed it to be consistent with `get_prompt`, it also raises `TemplateNotFoundError` when the prompt is not found.

The idea was that if someone makes a mistake in the name, they should get the error instead of silently returning the empty dict.

Reply:
What about
```py
        for pf in self._index.files:
            if pf.name == name and pf.version == version:
                if not pf.meta:
                    raise MetaNotFoundError
                return pf
        raise TemplateNotFoundError
```

Reply:
Makes sense :raised_hands:

Reply:
Actually, why not just `return self.get_prompt(...).meta`?